src/util/signature_utils.js

```diff
@@ -372,9 +372,34 @@ function authorize_request_account_by_token(token, requesting_account, is_secret
 }
 
 
+// In case the object access_keys in requesting_account has the is_active property
+// and is_active is false we will not continue to work on the request and immediately throw an error
+// access_key_id_to_find is a string (not SensitiveString)
+// inside requesting_account the access keys ids are SensitiveStrings
+function check_access_key_status(access_key_id_to_find, requesting_account) {
+    const access_keys = requesting_account.access_keys;
+    const index = _.findIndex(access_keys, item => item.access_key.unwrap() === access_key_id_to_find);
+    if (index < 0) {
+        // we should never get here
+        // in case we did, we need to debug the issue
+        dbg.warn(`check_access_key_status: could not find access_key_id ${access_key_id_to_find}`);
```

Comment:
so why are we just continuing? if this is not mean to be allowed it should throw in such a sensitive flow.

Reply:
I will change it and throw an error.